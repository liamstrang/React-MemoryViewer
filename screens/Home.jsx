import React from 'react';
import {KeyboardAvoidingView, Image, Text} from 'react-native';
import {Button} from 'react-native-paper';

/**
@author Liam Strang - 45426392

@description Screen to display the "Home" page. 
*/

//Homepage Styling
import styles from '../styles/homeStyles'

const Home = ({navigation}) => {
    return (
      <KeyboardAvoidingView style={styles.container} behavior="padding">
        <Image source={require('../assets/logo.png')} style={styles.image} />
        <Text style={styles.header}>Picture Collections</Text>
        <Text style={styles.subheader}>View and sort your favourite images!</Text>
        <Button style={styles.loginButton} icon="account-key" mode="contained" onPress={() => navigation.navigate("Login")}>
          Login
        </Button>
        <Button style={styles.registerButton} labelStyle={{color: '#004ae6'}} icon="account-plus" mode="outlined" onPress={() => navigation.navigate("Registration")}>
          Register
        </Button>
      </KeyboardAvoidingView>
    )
}

export default Home
