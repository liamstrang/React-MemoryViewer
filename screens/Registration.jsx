import React, {useState, useContext} from 'react';
import { View, KeyboardAvoidingView, Image, Text} from 'react-native';
import { TextInput as Input, Button} from 'react-native-paper';
import { AppContext } from '../services/appContext';

//Registration Styling
import styles from '../styles/registerStyles'

//Generate Images Function
import GenerateImages from '../services/generateImages';

/**
@author Liam Strang - 45426392

@description Screen to display the "Registration" page. 
*/

const Registration = ({navigation}) => {
  
  //Fetch all Users
  const UserProvider = useContext(AppContext)

  const [username, setusername] = useState('');
  const [password, setPassword] = useState('');

  const[signupError, setSignupError] = useState('')
  const[submitButton, setSubmitButton] = useState(false)

  const loginSubmit = () => {
    setSubmitButton(true);
    setTimeout(function() {
      if(UserProvider.users.find(u => u.username === username)){
        setSignupError("Username Already Exists")
      }else{
        UserProvider.users.push({"username": username, "password": password, "avatar": "https://robohash.org/"+username, "memories": GenerateImages(username)})
        navigation.navigate('Login')
      }
      setSubmitButton(false);
    }, 800);
  }

  return (
      <KeyboardAvoidingView style={styles.container} behavior="padding">
      <Image source={require('../assets/logo.png')} style={styles.image} />
      <Text style={styles.header}>Register</Text>
      <View style={styles.formContainer}>
      <Input
        style={styles.input}
        selectionColor='#004ae6'
        underlineColor="transparent"
        mode="outlined"
        label="Username"
        returnKeyType="next"
        value={username.value}
        onChangeText={text => setusername(text)}
        autoCapitalize="none"
        autoCompleteType="username"
        textContentType="username"
        keyboardType="default"
      />
      </View>
      <View style={styles.formContainer}>
      <Input
        style={styles.input}
        selectionColor='#004ae6'
        underlineColor="transparent"
        mode="outlined"
        label="Password"
        textContentType="password"
        returnKeyType="done"
        value={password.value}
        onChangeText={text => setPassword(text)}
        secureTextEntry
      />
      {signupError ? <Text style={styles.error}>{signupError}</Text> : null}
    </View>
    {submitButton ? <Button style={styles.button} loading={true} /> : <Button style={styles.button} icon="account-key" mode="contained" onPress={loginSubmit}>
        Register
    </Button>}
    </KeyboardAvoidingView>
  )
}

export default Registration
