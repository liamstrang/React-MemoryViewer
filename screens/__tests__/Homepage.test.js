import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home'
import Registration from '../Registration'
import Login from '../Login'

describe('Check Home Page', () => {
    it('Render Homepage', () => {
        const tree = renderer.create(<Home />).toJSON();
        expect(tree.children[1].children[0]).toBe('Picture Collections')
        expect(tree.children[2].children[0]).toBe('View and sort your favourite images!')
    })
    it('Homepage Styling', () => {
        const json = renderer.create(<Home/>).toJSON()
        expect(json.props.style[0].width).toBe('100%')
        expect(json.props.style[0].flex).toBe(1)
        expect(json.props.style[0].maxWidth).toBe(340)
        expect(json.props.style[0].alignSelf).toBe('center')
    })
})

describe('Check Register Page', () => {
    it('Render Register Page', () => {
        const tree = renderer.create(<Registration />).toJSON();
        expect(tree.children[2].type).toBe('View')
    })
    it('Render Register Image', () => {
        const tree = renderer.create(<Registration />).toJSON();
        expect(tree.children[0].props.source).toBe(1)
    })
    it('Register Styling', () => {
        const json = renderer.create(<Registration/>).toJSON()
        expect(json.props.style[0].width).toBe('100%')
        expect(json.props.style[0].flex).toBe(1)
        expect(json.props.style[0].maxWidth).toBe(340)
        expect(json.props.style[0].padding).toBe(20)
        expect(json.props.style[0].alignItems).toBe('center')
        expect(json.props.style[0].justifyContent).toBe('center')
    })
})

describe('Check Login Page', () => {
    it('Render Login Page', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree.children[2].type).toBe('View')
    })
    it('Render Login Image', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree.children[0].props.source).toBe(1)
    })
    it('Login Styling', () => {
        const json = renderer.create(<Login/>).toJSON()
        expect(json.props.style[0].width).toBe('100%')
        expect(json.props.style[0].flex).toBe(1)
        expect(json.props.style[0].maxWidth).toBe(340)
        expect(json.props.style[0].padding).toBe(20)
        expect(json.props.style[0].alignItems).toBe('center')
        expect(json.props.style[0].justifyContent).toBe('center')
    })
})