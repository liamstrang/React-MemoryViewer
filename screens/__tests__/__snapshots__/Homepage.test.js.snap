// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Snapshop 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
        "maxWidth": 340,
        "padding": 20,
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 128,
        "marginBottom": 12,
        "width": 128,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#004ae6",
        "fontSize": 26,
        "fontWeight": "bold",
        "paddingVertical": 14,
      }
    }
  >
    Picture Collections
  </Text>
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 16,
        "lineHeight": 26,
        "marginBottom": 14,
        "textAlign": "center",
      }
    }
  >
    View and sort your favourite images!
  </Text>
</View>
`;
